website/content/en/pre-docs/tasks/running-pods.md

```diff
@@ -0,0 +1,222 @@
+
+Examples below illustrate how to use Node affinity (`In`) and anti-affinity (`NotIn`).
+See [Node affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity) for details.
+When setting rules, the following Node affinity types define how hard or soft each rule is:
+
+* **requiredDuringSchedulingIgnoredDuringExecution**: This is a hard rule that must be met.
+* **preferredDuringSchedulingIgnoredDuringExecution**: This is a preference, but the pod can run on a node where it is not guaranteed.
+
+The `IgnoredDuringExecution` part of each tells the pod to keep running, even if conditions change on the node so the rules no longer matched.
+
+All examples below assume that the provisioner doesn't have constraints to prevent those zones from being used.
+The first constraint says you could use `us-west-2a` or `us-west-2b`, the second constraint makes it so only `us-west-2b` can be used.
+
+```
+ affinity:
+   nodeAffinity:
+     requiredDuringSchedulingIgnoredDuringExecution:
+       nodeSelectorTerms:
+         - matchExpressions:
+           - key: "topology.kubernetes.io/zone"
+             operator: "In"
+             values: ["us-west-2a, us-west-2b"]
+           - key: "topology.kubernetes.io/zone"
+             operator: "In"
+             values: ["us-west-2b"]
+```
+
+Changing the second operator to anti-affinity (`NotIn`) would allow the pod to run in `us-west-2a` only:
+
+```
+           - key: "topology.kubernetes.io/zone"
+             operator: "In"
+             values: ["us-west-2a, us-west-2b"]
+           - key: "topology.kubernetes.io/zone"
+             operator: "NotIn"
+             values: ["us-west-2b"]
+```
+
+Continuing to add to the example, nodeAffinity lets you match expressions so if one key-value pair doesn't work, you can go on to the next one.
+Here, if `us-west-2a` is not available, the pod can go to the East zone and run on a spot instance (notice the AWS-specific key).
+
+
+```
+ affinity:
+   nodeAffinity:
+     requiredDuringSchedulingIgnoredDuringExecution:
+       nodeSelectorTerms:
+         - matchExpressions: # OR
+           - key: "topology.kubernetes.io/zone" # AND
+             operator: "In"
+             values: ["us-west-2a, us-west-2b"]
+           - key: "topology.kubernetes.io/zone" # AND
+             operator: "NotIn"
+             values: ["us-west-2b"]
+         - matchExpressions: # OR
+           - key: "node.k8.aws/capacity-type" # AND
+             operator: "In"
+             values: ["spot"]
+           - key: "topology.kubernetes.io/zone" # AND
+             operator: "In"
+             values: ["us-east-2a"]
+```
+Karpenter will go through each of the matchExpressions in order and take the first one that works.
+However, if Karpenter fails to provision on the first matchExpression, it will delete it and try to use the second one.
+If they all fail, Karpenter will fail to provision for and deploy the pod.
```

Comment:
```suggestion
If they all fail, Karpenter will fail to provision the pod. Karpenter will backoff and retry over time, so if capacity becomes available, it will schedule the pod without user intervention.
```

Reply:
Done.